relayer/pkg/chainlink/txm/txm.go

```diff
@@ -143,24 +138,32 @@ func (txm *starktxm) run() {
 	}
 }
 
+const FEE_MARGIN uint64 = 115
+
 func (txm *starktxm) broadcastBatch(ctx context.Context, privKey, sender string, txs []types.Transaction) (txhash string, err error) {
 	// create new account
-	account, err := caigo.NewAccount(txm.curve, privKey, sender, txm.client)
+	account, err := caigo.NewAccount(privKey, sender, txm.client)
 	if err != nil {
 		return txhash, errors.Errorf("failed to create new account: %s", err)
 	}
 
 	// get fee for txm
-	fee, err := account.EstimateFee(ctx, txs)
+	fee, err := account.EstimateFee(ctx, txs, caigo.ExecuteDetails{})
 	if err != nil {
 		return txhash, errors.Errorf("failed to estimate fee: %s", err)
 	}
 
+	details := caigo.ExecuteDetails{
+		MaxFee: &types.Felt{
+			Int: new(big.Int).SetUint64((fee.OverallFee * FEE_MARGIN) / 100),
+		},
+	}
```

Comment:
This just duplicates the default internal logic: if you provide no max fee and don't call EstimateFee it will do the same thing internally. I inlined this since I figured we might want to weak the FEE_MARGIN a bit

Reply:
+1